CMakeLists.txt

```diff
@@ -318,7 +318,15 @@ else()
     usFunctionCheckCompilerFlags(-Wno-mismatched-tags US_CXX_FLAGS)
   endif()
 
-  usFunctionCheckCompilerFlags("-fvisibility=hidden -fvisibility-inlines-hidden" _have_visibility)
+  # Issue #56: Workaround a clang compiler issue (https://llvm.org/bugs/show_bug.cgi?id=26749).
+  # The typeinfo symbols do not have reliable symbol visibility. Instead of exporting all
+  # symbols only export the typeinfo symbols by default. See
+  # https://developer.apple.com/library/mac/technotes/tn2185/_index.html#//apple_ref/doc/uid/DTS10004200-CH1-SUBSECTION2
+  if(US_COMPILER_CLANG OR US_COMPILER_APPLE_CLANG)
```

Comment:
Only a few LLVM versions [have pre-built Mac OS X binaries](http://llvm.org/releases/download.html#3.7.0).

Thanks, I will modify the CMake code to handle both clang compilers only on Mac OS X.